packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+        event.localPosition,
+      );
+      _transformationController.value = _matrixTranslate(
+        _transformationController.value,
+        focalPointSceneScaled - focalPointScene,
+      );
+    }
+  }
+
+  // Handle inertia drag animation.
+  void _onAnimate() {
+    if (!_controller.isAnimating) {
+      _animation?.removeListener(_onAnimate);
+      _animation = null;
+      _controller.reset();
+      return;
+    }
+    // Translate such that the resulting translation is _animation.value.
+    final Vector3 translationVector = _transformationController.value.getTranslation();
+    final Offset translation = Offset(translationVector.x, translationVector.y);
+    final Offset translationScene = _transformationController.toScene(
+      translation,
+    );
+    final Offset animationScene = _transformationController.toScene(
+      _animation.value,
+    );
+    final Offset translationChangeScene = animationScene - translationScene;
+    _transformationController.value = _matrixTranslate(
+      _transformationController.value,
+      translationChangeScene,
+    );
+  }
+
+  void _onTransformationControllerChange() {
+    // A change to the TransformationController's value is a change to the
+    // state.
+    setState(() {});
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    _transformationController = widget.transformationController
+        ?? TransformationController();
+    _transformationController.addListener(_onTransformationControllerChange);
+    _controller = AnimationController(
+      vsync: this,
+    );
+  }
+
+  @override
+  void didUpdateWidget(InteractiveViewer oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.child != oldWidget.child || widget.boundaryMargin != oldWidget.boundaryMargin) {
+      _boundaryRectCached = null;
+    }
+    if (widget.transformationController != null
+        && widget.transformationController != _transformationController) {
+      _transformationController.removeListener(_onTransformationControllerChange);
+      _transformationController = widget.transformationController;
+      _transformationController.addListener(_onTransformationControllerChange);
+    }
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    _transformationController.removeListener(_onTransformationControllerChange);
+    if (widget.transformationController == null) {
+      _transformationController.dispose();
+    }
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget child = Transform(
+      transform: _transformationController.value,
+      child: KeyedSubtree(
+        key: _childKey,
+        child: widget.child,
+      ),
+    );
+
+    if (!widget.constrained) {
+      // These SingleChildScrollViews effectively remove the size constraints
+      // from the child. The child can even be bigger than the screen.
+      child = SingleChildScrollView(
```

Comment:
Are the nested scrollviews necessary? Can we achieve the same effect by wrapping it in an UnconstrainedBox?

Reply:
An UnconstrainedBox does seem to allow me to get the full size of the child, even beyond the size of the screen.  However, it complains about the child overflowing it.

I mentioned this before, but if I use an OverflowBox, it still constrains the size to the screen.  I just tried a couple permutations of both OverflowBox and UnconstrainedBox and didn't have any luck either.

Reply:
Any other ideas?

Reply:
ah, right I forgot about the overflow warnings that UnconstrainedBox generates. However, I'd expect that you can get the same behavior without the warnings by using an OverflowBox configured as unconstrained:

```dart
          ClipRect( // to clip the visual overflow
            child: OverflowBox(
              minWidth: 0.0,
              minHeight: 0.0,
              maxWidth: double.infinity,
              maxHeight: double.infinity,
              child: Container(
                width: 1000000,
                height: 1000000,
                color: Colors.green,
              ),
          );
```

But you're saying this doesn't size the Container to 1000000 x 1000000? Odd.

Reply:
I was missing the min/max sizing. This works, fixed in https://github.com/flutter/flutter/pull/56409/commits/49b329ce2aa51fc828864c795b755e253c5a1da6.